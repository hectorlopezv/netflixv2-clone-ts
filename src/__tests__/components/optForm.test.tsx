/* eslint-disable react/jsx-pascal-case */
import React from 'react';
import { render, fireEvent, getByTestId, getByAltText } from '@testing-library/react';


import OptForm, {OptForm_Input, OptForm_Button, OptForm_Text, OptForm_Break} from '../../components/optForm';


describe('<OptFrom />', () => {

    it('its render component <OptForm /> with populated data', () => {
        const {container, getByText, getByPlaceholderText} = render(
            <OptForm>
                <OptForm_Input placeholder="Email Address"/>
                <OptForm_Button>Try it now</OptForm_Button>
                <OptForm_Break/> 
                <OptForm_Text>
                    Ready to watch? enter your email 
                    to create or restart your membership
                </OptForm_Text>
            </OptForm>
        );
        //asertions --> dummy component

        expect(getByText('Try it now')).toBeTruthy();
        expect(getByText('Ready to watch? enter your email to create or restart your membership')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();
    });
});