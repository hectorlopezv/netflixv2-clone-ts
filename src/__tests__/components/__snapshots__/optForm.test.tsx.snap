// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptFrom /> its render component <OptForm /> with populated data 1`] = `
<div
  class="OptFrom__Container-sc-1yp3xvy-0 lowLVr"
>
  <input
    class="OptFrom__Input-sc-1yp3xvy-1 bxuxtm"
    placeholder="Email Address"
  />
  <button
    class="OptFrom__Button-sc-1yp3xvy-2 dSZZKp"
  >
    Try it now
    <img
      alt="Try Now"
      src="/images/icons/chevron-right.png"
    />
  </button>
  <div
    class="OptFrom__Break-sc-1yp3xvy-4 hwsWyN"
  />
  <p
    class="OptFrom__Text-sc-1yp3xvy-3 yDUth"
  >
    Ready to watch? enter your email to create or restart your membership
  </p>
</div>
`;
